src/vm/threadsuspend.cpp

```diff
@@ -6853,7 +6880,6 @@ void Thread::HijackThread(VOID *pvHijackAddr, ExecutionState *esb)
 
 #ifdef _DEBUG
     static int  EnterCount = 0;
-    _ASSERTE(EnterCount++ == 0);
```

Comment:
I would remove the `EnterCount` variable and all its uses. I do not think that these are a high-value assertions that needs to kept around.